exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java

```diff
@@ -128,55 +232,202 @@ void createTestKitWithoutServices() {
+        TransactionResult transactionResult = transactionResults.get(message.hash());
+        assertThat(transactionResult).isEqualTo(TransactionResult.successful());
+        return null;
+      });
+    }
+  }
 
-    @Override
-    public short getId() {
-      return SERVICE_ID;
+  @Test
+  void createBlockWithTransactions() {
+    try (TestKit testKit = TestKit.forService(TestServiceModule.class)) {
+      TransactionMessage message = constructTestTransactionMessage("Test message");
+      TransactionMessage message2 = constructTestTransactionMessage("Test message 2");
+
+      Block block = testKit.createBlockWithTransactions(ImmutableList.of(message, message2));
+      assertThat(block.getNumTransactions()).isEqualTo(2);
+
+      testKit.withSnapshot((view) -> {
+        checkTransactionsCommittedSuccessfully(view, block, message, message2);
+        return null;
+      });
     }
+  }
 
-    @Override
-    public String getName() {
-      return SERVICE_NAME;
+  @Test
+  void createBlockWithTransactionsVarargs() {
+    try (TestKit testKit = TestKit.forService(TestServiceModule.class)) {
+      TransactionMessage message = constructTestTransactionMessage("Test message");
+      TransactionMessage message2 = constructTestTransactionMessage("Test message 2");
+
+      Block block = testKit.createBlockWithTransactions(message, message2);
+      assertThat(block.getNumTransactions()).isEqualTo(2);
+
+      testKit.withSnapshot((view) -> {
+        checkTransactionsCommittedSuccessfully(view, block, message, message2);
+        return null;
+      });
     }
+  }
 
-    @Override
-    public Transaction convertToTransaction(RawTransaction rawTransaction) {
-      throw new UnsupportedOperationException();
+  private TransactionMessage constructTestTransactionMessage(String payload) {
+    return TransactionMessage.builder()
+        .serviceId(TestService.SERVICE_ID)
+        .transactionId(TestTransaction.ID)
+        .payload(payload.getBytes(BODY_CHARSET))
+        .sign(KEY_PAIR, CRYPTO_FUNCTION);
+  }
+
+  private void checkTransactionsCommittedSuccessfully(
+      View view, Block block, TransactionMessage message, TransactionMessage message2) {
+    Blockchain blockchain = Blockchain.newInstance(view);
+    assertThat(blockchain.getHeight()).isEqualTo(1);
+    assertThat(block).isEqualTo(blockchain.getBlock(1));
+    Map<HashCode, TransactionResult> transactionResults = toMap(blockchain.getTxResults());
+    assertThat(transactionResults).hasSize(2);
+
+    TransactionResult transactionResult = transactionResults.get(message.hash());
+    assertThat(transactionResult).isEqualTo(TransactionResult.successful());
+    TransactionResult transactionResult2 = transactionResults.get(message2.hash());
+    assertThat(transactionResult2).isEqualTo(TransactionResult.successful());
+  }
+
+  @Test
+  void createBlockWithTransactionWithWrongServiceId() {
+    Class<RuntimeException> exceptionType = RuntimeException.class;
```

Comment:
(1) The thrown exception type is on the wrong abstraction level, RuntimeException communicates nothing. Since it is a testkit,
 I would expect either IllegalArgumentException or https://ota4j-team.github.io/opentest4j/docs/current/api/org/opentest4j/AssertionFailedError.html
(2) As we established in the requirements, the error messages must be user-friendly and tell exactly what
the problem is (or which problems are possible). In this case I think there are two possible problems:
   - A test developer passed an incorrectly serialized transaction (they intended to pass a proper one).
     The implementation of the service is fine.
   - A test dev passed a correctly serialized transaction, but the implementation of the service
     (particularly, TransactionConverter) does not handle that transaction correctly.

In both cases, the actual thrown exception *and* the details about the transaction message are quite useful
to understand what is broken.

Speaking of the implementation, it might be more convenient to check that each tx message is correct and create
a proper exception _in Java_, before passing them to the native code.

Reply:
There might be a third case which we do not explicitly support — intentionally passing an invalid transaction. I think we shall either recommend testing that separately from the testkit (just take your TxConverter); or add first-class support (checkCannotDeserialize?). I think the first option is fine for now.